### ISSUE TITLE
Inaccuracy in the "Constants" section

### ISSUE THREAD

**Issue by Retsam** [hemanth/functional-programming-jargon#55]
> With the above two constants the following expression will always return true.

> ```john.age + five === ({name: 'John', age: 30}).age + (5)```

This above statement isn't true; while `john` is declared as a constant; that only means that `john` can't be replaced with a different object, not that its fields can't be modified.  A simple `john.age = 10` would cause the above expression to return false, despite the definition of `john` as a constant.

**Comment by jethrolarson** [comment]
Yeah there's definitely common confusion about const not making objects immutable. This section should probably distinguish between them

**Comment by joe-strummer** [comment]
Yeah just came here to say that - a more straightforward example could just be to have `const five` and `const six` for instance.

**Comment by isiahmeadows** [comment]
I think it could be fixed by saying it's referentially transparent for *non-symbol primitives* (well...except for `Symbol.for("foo")`).
